RestaurantReservation.API/Profiles/EmployeeProfile.cs

```diff
@@ -0,0 +1,16 @@
+﻿using AutoMapper;
+using RestaurantReservation.API.Models;
+using RestaurantReservation.Db.Entities;
+
+namespace RestaurantReservation.API.Profiles
+{
+    public class EmployeeProfile : Profile
+    {
+        public EmployeeProfile()
+        {
+            CreateMap<Employee, EmployeeDTO>();
```

Comment:
We usually do not expose DB models to the API layer, if you want to map these models, profiles need to exist in the `Application` layer, and you can have these models in the `Application` layer instead of the API layer.

And to make that happen, we need to have API project to reference Application project, and DB project to reference Application project.

Lets actually need the new project Domain not Application. The idea here is that API and DB projects are frameworks/details that the domain project shouldn't rely on, it should be the way around, since domain layer usually refers to the business logic that should be pretty stable and won't require much changes in the future.

Having Domain to rely on DB layer for example is not a good idea, since DB is not stable as you could change the implementation of DB and use a different framework and Domain then will need to be updated as well. Domain should be stable and not affected as it's the business core of the project.

Read more about this, it's called Clean Architecture.

Reply:
Also, I see that DB models are returned from the repos, and the mapping happens in the API layer. As mentioned above, we don't expose DB models to the other layer and should be `internal` in the DB project. Instead, the repos should return domain/dto models,  and the mapping should happen from within the repos.

This mean the repo should only accept and return domain models/dtos, not db models.